ClosedXML_Tests/Excel/CalcEngine/MathTrigTests.cs

```diff
@@ -447,6 +447,72 @@ public void Base_ReturnsCorrectResultOnInputWithMinimalLength(int input, int the
             Assert.AreEqual(expectedResult, actual);
         }
 
+        [Theory]
```

Comment:
Theory? What does that mean?

Reply:
A Theory is more than a fact, it's something like a general rule: For some pattern of inputs a rule appllies.
A [Fact] checks a single fact once (the method has no parameters and is executed once during test execution).

A [Theory] checks a rule for a set of input parameter sets.
In NUnit the [TestCase] attribute is in fact some kind of theory itself, but with [TestCase] you have to specify any test case individually.
Here the test data can be generated as any n greater or equal than 0 (in theory ;) ) is a valid input. 

In practise we have two restrictions here:

- the current implementation is restricted to Int64/long  as a result type, while Excel can cope with bigger integer values (I guess Microsoft uses double for any calculation in background), so we can't use too big inputs currently
-  with more test cases generated e.g. by the range attribute test execution time increases, so I'm careful to use a too big range here.

If you're reading through theoretical stuff about testing, using theories and with them more some kind of "general rules" is sometimes also referred to as property based testing (where the rule I referred to above is called a property), from mathmatics background you may know these rules under the term "invariant".